tests/app/views/test_suppliers.py

```diff
@@ -1280,13 +1280,23 @@ def setup(self):
                 data=json.dumps({'updated_by': 'interested@example.com'}),
                 content_type='application/json')
 
+            # framework_agreement_version gets saved into the agreementDetails JSON blob of a supplier_framework
+            # when an agreement is returned
+            Framework.query.filter_by(slug='digital-outcomes-and-specialists').update(
```

Comment:
Oh this is interesting... not all frameworks will have a `framework_agreement_version`.  In fact, only G-Cloud 8 will.

These "return agreement" routes all still need to work for older frameworks that don't have any framework agreement details, where there won't be any framework agreement details to refer to.

So (sorry!) it might be nice to leave the old-style returning agreements flow tests exactly as they were - to prove that the existing flow isn't broken - and then have a separate set of G-Cloud 8 returning agreements tests where there are framework agreement details.  These new tests should check the new agreement details stuff.